File: task_d_evolve_api/order_handler.py
from notification_service import NotificationService


notifier = NotificationService()


def place_order(user: str, item: str):
    """Process a new order and notify the user."""
    print(f"Processing order for {item}...")
    notifier.send(user, f"Your order for {item} has been placed.")


def cancel_order(user: str, order_id: str):
    """Cancel an existing order and notify the user."""
    print(f"Cancelling order {order_id}...")
    notifier.send(user, f"Your order {order_id} has been cancelled.")

File: TS/task_d_evolve_api/orderHandler.ts
import { NotificationService } from "./notificationService";

const notifier = new NotificationService();

export function placeOrder(user: string, item: string): void {
  /** Process a new order and notify the user. */
  console.log(`Processing order for ${item}...`);
  notifier.send(user, `Your order for ${item} has been placed.`);
}

export function cancelOrder(user: string, orderId: string): void {
  /** Cancel an existing order and notify the user. */
  console.log(`Cancelling order ${orderId}...`);
  notifier.send(user, `Your order ${orderId} has been cancelled.`);
}
